src/data-workspace/use-data-value-mutation/use-set-data-value-comment-mutation.js

```diff
@@ -0,0 +1,38 @@
+import { useApiAttributeParams } from '../../shared/index.js'
+import optimisticallySetDataValueComment from './optimistically-set-data-value-comment.js'
+import useDataValueMutation from './use-data-value-mutation.js'
+import useMutationFn from './use-mutation-fn.js'
+
+export const SET_DATA_VALUE_COMMENT_MUTATION = {
+    resource: 'dataValues',
+    type: 'create',
+    data: ({ co, de, ds, ou, pe, comment }) => {
+        return {
+            // Easier to read this way..
+            // eslint-disable-next-line
+            co, de, ds, ou, pe,
+
+            // when sending `undefined` something causes that
+            // to get stored as a string `"undefined"`
+            comment: comment ? comment : '',
+        }
+    },
+}
+
+export function useSetDataValueCommentMutation(onDone) {
+    const { attributeCombo, attributeOptions } = useApiAttributeParams()
+    const mutationFn = useMutationFn(SET_DATA_VALUE_COMMENT_MUTATION)
+
+    return useDataValueMutation({
+        mutationFn,
+        onSuccess: onDone,
+        onMutate: ({ queryClient, newDataValue, dataValueSetQueryKey }) =>
+            optimisticallySetDataValueComment({
+                queryClient,
+                newDataValue,
```

Comment:
Regarding comment above. You could format the value of `newDataValue` here. Could even create different functions to handle the transformation, and then use a shared `optimisticallySetDataValue`-function.

Reply:
I don't understand this exactly. So in my head this would look like this:

1. The individual hooks format the next value. In order to do so, they need to know whether it's a new value or an update to an existing value
2. The hook then passes the updated value and a boolean whether this was an update or not to the optimistically update function

We'll end up with
* two different optimistic update functions (essentially "optimistically update" and "optimistically delete")
* either four new functions for formatting the slightly different values OR we'll add this to the hooks without exacting into a function

We'd either end up with more functions or we'd have optimistic update related code in two places (the hook and the helper), which seems worse that the current implementation (as there are four optimistic update related places for four slightly different update scenarios).

Maybe you meant something different though?